_posts/2020-12-02-pipelined-region-sheduling.md

```diff
@@ -0,0 +1,284 @@
+There is no way to run the job with fewer resources than needed to run the largest region. Hence, there can be no resource waste.
+
+<div class="alert alert-info" markdown="1">
+<span class="label label-info" style="display: inline-block"><span class="glyphicon glyphicon-info-sign" aria-hidden="true"></span> Note (out of scope)</span>
+The amount of resources, required to run a region, can be further optimized separately.
+It depends on _co-location constraints_ and _slot sharing groups_ of the region’s _subtasks_.
+Check [FLINK-18689](https://issues.apache.org/jira/browse/FLINK-18689) for details.
+</div>
+
+# Conclusion
+
+Scheduling is a fundamental component of the Flink stack. In this blogpost, how scheduling affects resource utilization and failover as a part of the user experience.
+We described the limitations of Flink’s old scheduler and introduced a new approach to tackle them: the  _pipelined region scheduler_, which will be available with Flink 1.12.
+The blogpost also explained how the new _pipelined region failover_ works, which has been introduced in Flink 1.11.
+
+Stay tuned for more additions and features in upcoming releases. If you have any suggestions or questions for the community,
+we encourage you to sign up to the Apache Flink [mailing lists](https://flink.apache.org/community.html#mailing-lists) and become part of the discussion.
+
+# Appendix
+
+## What is scheduling?
+
+### ExecutionGraph
+
+A _job_, written against Flink API, is a pipeline of connected _operators_ to process data.
+Together the operators form a _[JobGraph](https://ci.apache.org/projects/flink/flink-docs-release-1.11/internals/job_scheduling.html#jobmanager-data-structures)_.
+Each _operator_ has a certain number of _subtasks_ executed in parallel. The _subtask_ is the actual execution unit in Flink.
+Each subtask can consume user records from other subtasks (inputs), process them and produce records for further consumption by other _subtasks_ (outputs) down the stream.
+There are _source subtasks_ without inputs and _sink subtasks_ without outputs. Hence, the _subtasks_ form the nodes of the
+_[ExecutionGraph](https://ci.apache.org/projects/flink/flink-docs-release-1.11/internals/job_scheduling.html#jobmanager-data-structures)_.
+
+
+### Intermediate results
+
+There are also two major data-exchange types to produce and consume results by _operators_ and their _subtasks_: _pipelined_ and _blocking_.
+They are basically types of the edges in the _ExecutionGraph_.
+
+The _pipelined_ result can be consumed record by record. It means that the consumer can already run once the first result records have been produced.
+The _pipelined_ result can be a never ending output of records, e.g. in case of a _streaming job_.
+
+The _blocking_ result can be consumed only when its _production_ is done. Hence, the _blocking_ result is always finite
+and the consumer of the _blocking_ result can run only when the producer has finished its execution.
```

Comment:
```suggestion
A _pipelined_ result can be consumed record by record. This means that the consumer can already run once the first result records have been produced.
A _pipelined_ result can be a never ending output of records, e.g. in case of a _streaming job_.

A _blocking_ result can be consumed only when its _production_ is done. Hence, the _blocking_ result is always finite
and the consumer of the _blocking_ result can run only when the producer has finished its execution.
```